pkg/volume/csi/csi_attacher.go

```diff
@@ -303,11 +303,9 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		volDataKey.driverName: csiSource.Driver,
 	}
 	if err = saveVolumeData(dataDir, volDataFileName, data); err != nil {
-		klog.Error(log("failed to save volume info data: %v", err))
-		if cleanErr := os.RemoveAll(dataDir); cleanErr != nil {
-			klog.Error(log("failed to remove dir after error [%s]: %v", dataDir, cleanErr))
-		}
-		return err
+		errMsg := log("failed to save volume info data: %v", err)
+		klog.Error(errMsg)
+		return errors.New(errMsg)
```

Comment:
similar comment as below

Reply:
removeMountDir() will be called in defer func here.

Reply:
@fankangbest Can you please address this comment? I am happy to have unit test merged later on but we need this fixed before we can merge this.

Reply:
It looks like this still needs to be addressed in `csi_plugin.go`, but the attacher code appears to have already been updated in master:

https://github.com/kubernetes/kubernetes/blob/227aa51b942a3b76bf237bb5508951a351b043b4/pkg/volume/csi/csi_attacher.go#L307-L309